### PR TITLE
Update CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by BTernaryTau**
Based on my onboarding experience, I made two additions to the contributing page.

1. I added a mention of the metamath-lamp tool. I found that having a proof assistant that didn't need to be installed made trying my hand at writing metamath proofs much easier and more welcoming. Thus, I think it should be mentioned on the contributing page.
2. I mentioned that running the minimize script is encouraged. [This change was discussed in a thread from several years ago](https://groups.google.com/g/metamath/c/3QEuZYbjc38/m/zSACd_TpBwAJ), but was never actually done. I didn't even realize this tool existed until after I submitted several proofs, so I think it would be very helpful to include it as well.

In addition, I updated the mmj2 link to point to the mmj2 section of the website's main page rather than the book section.

### CODE REVIEW

**Review by david-a-wheeler — APPROVED** [review]

**Comment by david-a-wheeler** [comment]
Thanks! It's great to have onboarding feedback from people who've only recently done it. It's sometimes easy to forget stuff like this.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -23,8 +23,11 @@ Walking through the
 To create a proof, you need to pick a tool.
 The Metamath system specifies a common file format, not a single tool
 everyone has to use.
-Many in the community use the [mmj2 tool](http://us.metamath.org#book), and
+Many in the community use the [mmj2 tool](http://us.metamath.org#mmj2), and
 many of the rest use the original metamath C program (aka "metamath.exe").
+A third option is the [metamath-lamp tool](https://lamp-guide.metamath.org/),
+which can be accessed directly from your web browser at
+[https://expln.github.io/lamp/latest/index.html](https://expln.github.io/lamp/latest/index.html).
```
Comment (icecream17) [inline CONTRIBUTING.md:30]:
```suggestion
<https://expln.github.io/lamp/latest/index.html>.
```

merely a shortening, the button to apply suggestion should work

**Review by icecream17 — APPROVED** [review]

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -107,6 +110,12 @@ adjust the proof indentation to match any indentation changes made by /rewrap.
 Then, 'verify markup' will check that no lines became too long due to different
 indentation.  Finally, 'verify proof' is there because you might as well.
 
+In addition, you are encouraged to run `scripts/minimize label`, where label is
+the name of your proof. This script attempts to to shorten the proof using other
```
Comment (avekens) [inline CONTRIBUTING.md:114]: Typo: to to

Comment (BTernaryTau) [reply CONTRIBUTING.md]: Thanks, fixed.

Comment (avekens) [reply CONTRIBUTING.md]: Maybe " name of your theorem" instead of  "name of your proof".

**Comment by avekens** [comment]
Please ignore the link above - I do not know how github inserted it!

**Comment by BTernaryTau** [comment]
> Please ignore the link above - I do not know how github inserted it!

I don't have permission to remove the linked issue. Would someone else be able to remove the link so this PR doesn't close an unrelated issue?

**Comment by avekens** [comment]
> > Please ignore the link above - I do not know how github inserted it!
> 
> I don't have permission to remove the linked issue. Would someone else be able to remove the link so this PR doesn't close an unrelated issue?

OK, I could have removed it by myself (I did not know about this feature, available at the right side, and probably I just clicked there unintentionally).

**Comment by jkingdon** [comment]
> > > Please ignore the link above - I do not know how github inserted it!
> > 
> > 
> > I don't have permission to remove the linked issue. Would someone else be able to remove the link so this PR doesn't close an unrelated issue?
> 
> OK, I could have removed it by myself (I did not know about this feature, available at the right side, and probably I just clicked there unintentionally).

Possible but it can also happen by typing "fix #xxxx" in a pull request or maybe other mechanisms I don't know about.

Thanks for pointing out that right sidebar, that is easier than the workarounds I have been using when I trigger this accidentally.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -107,6 +110,12 @@ adjust the proof indentation to match any indentation changes made by /rewrap.
 Then, 'verify markup' will check that no lines became too long due to different
 indentation.  Finally, 'verify proof' is there because you might as well.
 
+In addition, you are encouraged to run `scripts/minimize label`, where label is
+the name of your theorem. This script attempts to shorten the proof using other
+theorems in the database. Over many proofs, this can make a big difference in
+speeding up various functions, such as loading the database into a tool.
+However, this step is not a requirement for submissions to be accepted.
```
Comment (jkingdon) [inline CONTRIBUTING.md:117]: Perhaps "requirement" would be overstating it but we do, if I understand correctly, encourage this. The text in the linked thread which says "In the past we have occasionally run the entire [set.mm](http://set.mm/) through minimization, and that will probably happen again at some point in the future when there is a volunteer with the necessary CPU resources" has only become more true as set.mm has grown and the volunteer/CPU resources available for mass minimizations seem to have been even more scarce than when that was written.

```suggestion
We encourage doing this before submitting proofs.
```

Comment (BTernaryTau) [reply CONTRIBUTING.md:117]: >  So we encourage (but do not require) that people minimize their proofs when it is not too much trouble to do so.

I tried to keep this consistent with the above policy as stated in the thread by mentioning both that it is encouraged (at the start of the paragraph) and that it is not required. That said, I'm fine with removing the last sentence so long as there's a consensus that we don't want it.

Comment (jkingdon) [reply CONTRIBUTING.md:117]: Oh good point about the start of the paragraph already saying encouraged. I suppose I'd drop the last sentence and keep the rest but I guess it doesn't matter a whole lot to me.

**Review by jkingdon — APPROVED** [review]

**Review by avekens — APPROVED** [review]

**Pull request merged**